harness/scripts/docker/builder.sh.twig

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+function builder_name() {
+  local namespace="{{ @('namespace') }}"
+  if [[ $namespace == "" ]]; then
+    namespace="unknown"
+  fi
+
+  echo "$namespace-{{ @('app.version') }}"
```

Comment:
(just redoing the comment as a PR comment so can be marked resolved later)

namespace IS app.version, so you're just repeating it twice :) https://github.com/inviqa/harness-docker/blob/main/harness/attributes/environment/pipeline.yml#L3

in the gitlab stuff though we set it to the pipeline id so that concurrent builds of the same git ref (I know, why?!) don't interfere

Reply:
> hmm, yeah in pipeline the namespace is just the git ref, but locally it's the same as compose project name, which is probably what we want in pipeline, e.g. ci-go-sample-8db5ae9de046ea6f24550d0f0f1a45bd3389664d

If that is the case, we would also want the docker compose project name to be it as well, which currently is just namespace

Reply:
so if you want the project name in it (which I think is good for docker compose project name as well, as SSHing into gitlab runners and having to find the project name is hard), then I think it should be https://github.com/inviqa/harness-docker/blob/main/harness/attributes/environment/pipeline.yml#L3 updated to include the workspace.name instead

Reply:
tho could be another PR to do that (we'd prob also want to use the short sha ref as well so to avoid making it too long)

Reply:
that's a good idea, I will do that in a follow-up PR 👍🏻

Reply:
so as for here, just @('namespace'), which you don't need to check existence of, and possibly the function/script is unnecessary

Reply:
> actually, I'll do it in this PR as this is not in a working-state for projects yet, ...

a project with a number for a first letter name will still be affected :)

Reply:
removed the additional script, and just using namespace for now, will sort out namespace in a follow-up PR